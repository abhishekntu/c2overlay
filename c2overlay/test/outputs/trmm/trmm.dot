digraph graphname {
N55 [color="black", label="store_Imm_8_N81_N55"];
N54 [color="black", label="N37_mul_N37_N54"];
N58 [color="black", label="store_Imm_2_N75_N58"];
N49 [color="black", label="N30N44_mul_N30_add_N44_N49"];
N48 [color="black", label="N72_mul_N72_N48"];
N63 [color="black", label="store_Imm_4_N79_N63"];
N62 [color="black", label="store_Imm_1_N78_N62"];
N61 [color="black", label="store_Imm_7_N73_N61"];
N60 [color="black", label="store_Imm_6_N77_N60"];
N41 [color="black", label="N24_mul_Imm_9_N24_N41"];
N40 [color="black", label="N27N25_mul_N27_add_N25_N40"];
N43 [color="black", label="N70_mul_N70_N43"];
N42 [color="black", label="N49N32_mul_N49_add_N32_N42"];
N45 [color="black", label="N50_mul_N50_N45"];
N44 [color="black", label="N57N71_mul_N57_add_N71_N44"];
N47 [color="black", label="N38_mul_Imm_9_N38_N47"];
N46 [color="black", label="N51_mul_N51_N46"];
N29 [color="black", label="N40_mul_N40_N29"];
N28 [color="black", label="N41N65_mul_N41_add_N65_N28"];
N59 [color="black", label="store_Imm_0_N76_N59"];
N23 [color="black", label="N47N35_mul_N47_add_N35_N23"];
N22 [color="black", label="N54N64_mul_N54_add_N64_N22"];
N21 [color="black", label="N53_mul_Imm_9_N53_N21"];
N20 [color="black", label="N52N21_mul_N52_add_N21_N20"];
N27 [color="black", label="N61N56_mul_N61_add_N56_N27"];
N26 [color="black", label="N62N63_mul_N62_add_N63_N26"];
N25 [color="black", label="N67N66_mul_N67_add_N66_N25"];
N24 [color="black", label="N46N60_mul_N46_add_N60_N24"];
N8 [color="black", label="load_Imm_1_N9_N8"];
N9 [color="black", label="load_Imm_1_N11_N9"];
N1 [color="black", label="load_Imm_4_N16_N1"];
N2 [color="black", label="load_Imm_8_N12_N2"];
N3 [color="black", label="load_Imm_6_N17_N3"];
N4 [color="black", label="load_Imm_2_N14_N4"];
N5 [color="black", label="load_Imm_0_N13_N5"];
N6 [color="black", label="load_Imm_7_N10_N6"];
N7 [color="black", label="load_Imm_7_N8_N7"];
N12 [color="black", label="load_Imm_3_N3_N12"];
N13 [color="black", label="load_Imm_3_N4_N13"];
N10 [color="black", label="load_Imm_5_N1_N10"];
N11 [color="black", label="load_Imm_5_N2_N11"];
N16 [color="black", label="load_Imm_0_N7_N16"];
N17 [color="black", label="load_Imm_4_N18_N17"];
N14 [color="black", label="load_Imm_8_N5_N14"];
N15 [color="black", label="load_Imm_2_N6_N15"];
N56 [color="black", label="store_Imm_5_N80_N56"];
N57 [color="black", label="store_Imm_3_N74_N57"];
N18 [color="black", label="load_Imm_6_N15_N18"];
N19 [color="black", label="N55_mul_Imm_9_N55_N19"];
N52 [color="black", label="N34_mul_Imm_9_N34_N52"];
N53 [color="black", label="N36_mul_Imm_9_N36_N53"];
N50 [color="black", label="N31_mul_N31_N50"];
N51 [color="black", label="N33N19_mul_N33_add_N19_N51"];
N38 [color="black", label="N22N26_mul_N22_add_N26_N38"];
N39 [color="black", label="N20_mul_Imm_9_N20_N39"];
N30 [color="black", label="N42N39_mul_N42_add_N39_N30"];
N31 [color="black", label="N45_mul_N45_N31"];
N32 [color="black", label="N69N58_mul_N69_add_N58_N32"];
N33 [color="black", label="N68_mul_Imm_9_N68_N33"];
N34 [color="black", label="N29N43_mul_N29_add_N43_N34"];
N35 [color="black", label="N28_mul_N28_N35"];
N36 [color="black", label="N59N48_mul_N59_add_N48_N36"];
N37 [color="black", label="N23_mul_Imm_9_N23_N37"];
N54 -> N28;
N49 -> N39;
N48 -> N26;
N41 -> N58;
N40 -> N37;
N43 -> N32;
N42 -> N30;
N45 -> N42;
N44 -> N36;
N47 -> N57;
N46 -> N44;
N29 -> N25;
N28 -> N51;
N23 -> N47;
N22 -> N41;
N21 -> N61;
N20 -> N21;
N27 -> N19;
N26 -> N22;
N25 -> N20;
N24 -> N23;
N8 -> N24;
N8 -> N42;
N8 -> N35;
N9 -> N42;
N9 -> N43;
N9 -> N48;
N1 -> N28;
N1 -> N43;
N1 -> N25;
N2 -> N40;
N2 -> N36;
N2 -> N22;
N3 -> N46;
N3 -> N29;
N3 -> N34;
N4 -> N27;
N4 -> N30;
N4 -> N22;
N5 -> N32;
N5 -> N26;
N5 -> N45;
N6 -> N44;
N6 -> N48;
N6 -> N38;
N7 -> N44;
N7 -> N25;
N7 -> N35;
N12 -> N50;
N12 -> N31;
N12 -> N54;
N13 -> N32;
N13 -> N54;
N13 -> N29;
N10 -> N51;
N10 -> N40;
N10 -> N23;
N11 -> N20;
N11 -> N51;
N11 -> N27;
N16 -> N34;
N16 -> N45;
N16 -> N31;
N17 -> N38;
N17 -> N24;
N17 -> N28;
N14 -> N20;
N14 -> N36;
N14 -> N49;
N15 -> N23;
N15 -> N30;
N15 -> N49;
N18 -> N26;
N18 -> N50;
N18 -> N46;
N19 -> N62;
N52 -> N63;
N53 -> N55;
N50 -> N38;
N51 -> N52;
N38 -> N40;
N39 -> N60;
N30 -> N33;
N31 -> N24;
N32 -> N27;
N33 -> N59;
N34 -> N49;
N35 -> N34;
N36 -> N53;
N37 -> N56;
}
