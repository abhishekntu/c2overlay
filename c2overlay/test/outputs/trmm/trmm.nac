procedure foo()
{
  localvar s32 D_2408;
  localvar s32 D_2409;
  localvar s32 D_2410;
  localvar s32 D_2411;
  localvar s32 D_2412;
  localvar s32 D_2413;
  localvar s32 D_2414;
  localvar s32 D_2415;
  localvar s32 D_2416;
  localvar s32 D_2417;
  localvar s32 D_2418;
  localvar s32 D_2419;
  localvar s32 D_2420;
  localvar s32 D_2421;
  localvar s32 D_2422;
  localvar s32 D_2423;
  localvar s32 D_2424;
  localvar s32 D_2425;
  localvar s32 D_2426;
  localvar s32 D_2427;
  localvar s32 D_2428;
  localvar s32 D_2429;
  localvar s32 D_2430;
  localvar s32 D_2431;
  localvar s32 D_2432;
  localvar s32 D_2433;
  localvar s32 D_2434;
  localvar s32 D_2435;
  localvar s32 D_2436;
  localvar s32 D_2437;
  localvar s32 D_2438;
  localvar s32 D_2439;
  localvar s32 D_2440;
  localvar s32 D_2441;
  localvar s32 D_2442;
  localvar s32 D_2443;
  localvar s32 D_2444;
  localvar s32 D_2445;
  localvar s32 D_2446;
  localvar s32 D_2447;
  localvar s32 D_2448;
  localvar s32 D_2449;
  localvar s32 D_2450;
  localvar s32 D_2451;
  localvar s32 D_2452;
  localvar s32 D_2453;
  localvar s32 D_2454;
  localvar s32 D_2455;
  localvar s32 D_2456;
  localvar s32 D_2457;
  localvar s32 D_2458;
  localvar s32 D_2459;
  localvar s32 D_2460;
  localvar s32 D_2461;
  localvar s32 D_2462;
  localvar s32 D_2463;
  localvar s32 D_2464;
  localvar s32 D_2465;
  localvar s32 D_2466;
  localvar s32 D_2467;
  localvar s32 D_2468;
  localvar s32 D_2469;
  localvar s32 D_2470;
  localvar s32 D_2471;
  localvar s32 D_2472;
  localvar s32 D_2473;
  localvar s32 D_2474;
  localvar s32 D_2475;
  localvar s32 D_2476;
  localvar s32 D_2477;
  localvar s32 D_2478;
  localvar s32 D_2479;
  localvar s32 A[9];
  localvar s32 B[9];
  localvar s32 C[9];
  localvar s32 alpha;
  localvar s32 i;
  localvar s32 j;
  localvar s32 t00000;
  localvar s32 t00001;
  localvar s32 t00002;
  localvar s32 t00003;
  localvar s32 t00004;
  localvar s32 t00005;
  localvar s32 t00006;
  localvar s32 t00007;
  localvar s32 t00008;
  localvar s32 t00009;
  localvar s32 t00010;
  localvar s32 t00011;
  localvar s32 t00012;
  localvar s32 t00013;
  localvar s32 t00014;
  localvar s32 t00015;
  localvar s32 t00016;
  localvar s32 t00017;
  localvar s32 t00018;
  localvar s32 t00019;
  localvar s32 t00020;
  localvar s32 t00021;
  localvar s32 t00022;
  localvar s32 t00023;
  localvar s32 t00024;
  localvar s32 t00025;
  localvar s32 t00026;
  localvar s32 t00027;
  localvar s32 t00028;
  localvar s32 t00029;
  localvar s32 t00030;
  localvar s32 t00031;
  localvar s32 t00032;
  localvar s32 t00033;
  localvar s32 t00034;
  localvar s32 t00035;
  localvar s32 t00036;
  localvar s32 t00037;
  localvar s32 t00038;
  localvar s32 t00039;
  localvar s32 t00040;
  localvar s32 t00041;
  localvar s32 t00042;
  localvar s32 t00043;
  localvar s32 t00044;
  localvar s32 t00045;
  localvar s32 t00046;
  localvar s32 t00047;
  localvar s32 t00048;
  localvar s32 t00049;
  localvar s32 t00050;
  localvar s32 t00051;
  localvar s32 t00052;
  localvar s32 t00053;
  localvar s32 t00054;
  localvar s32 t00055;
  localvar s32 t00056;
  localvar s32 t00057;
  localvar s32 t00058;
  localvar s32 t00059;
  localvar s32 t00060;
  localvar s32 t00061;
  localvar s32 t00062;
L0005:
  alpha <= ldc 9;
  D_2408 <= load A,0;
  D_2409 <= load B,0;
  D_2410 <= mul D_2408,D_2409;
  D_2411 <= load A,1;
  D_2412 <= load B,1;
  D_2413 <= mul D_2411,D_2412;
  D_2414 <= add D_2410,D_2413;
  D_2415 <= load A,2;
  D_2416 <= load B,2;
  D_2417 <= mul D_2415,D_2416;
  D_2418 <= add D_2414,D_2417;
  D_2419 <= mul D_2418,alpha;
  C <= store D_2419,0;
  D_2408 <= load A,0;
  D_2420 <= load B,3;
  D_2421 <= mul D_2408,D_2420;
  D_2411 <= load A,1;
  D_2422 <= load B,4;
  D_2423 <= mul D_2411,D_2422;
  D_2424 <= add D_2421,D_2423;
  D_2415 <= load A,2;
  D_2425 <= load B,5;
  D_2426 <= mul D_2415,D_2425;
  D_2427 <= add D_2424,D_2426;
  D_2428 <= mul D_2427,alpha;
  C <= store D_2428,1;
  D_2408 <= load A,0;
  D_2429 <= load B,6;
  D_2430 <= mul D_2408,D_2429;
  D_2411 <= load A,1;
  D_2431 <= load B,7;
  D_2432 <= mul D_2411,D_2431;
  D_2433 <= add D_2430,D_2432;
  D_2415 <= load A,2;
  D_2434 <= load B,8;
  D_2435 <= mul D_2415,D_2434;
  D_2436 <= add D_2433,D_2435;
  D_2437 <= mul D_2436,alpha;
  C <= store D_2437,2;
  D_2438 <= load A,3;
  D_2409 <= load B,0;
  D_2439 <= mul D_2438,D_2409;
  D_2440 <= load A,4;
  D_2412 <= load B,1;
  D_2441 <= mul D_2440,D_2412;
  D_2442 <= add D_2439,D_2441;
  D_2443 <= load A,5;
  D_2416 <= load B,2;
  D_2444 <= mul D_2443,D_2416;
  D_2445 <= add D_2442,D_2444;
  D_2446 <= mul D_2445,alpha;
  C <= store D_2446,3;
  D_2438 <= load A,3;
  D_2420 <= load B,3;
  D_2447 <= mul D_2438,D_2420;
  D_2440 <= load A,4;
  D_2422 <= load B,4;
  D_2448 <= mul D_2440,D_2422;
  D_2449 <= add D_2447,D_2448;
  D_2443 <= load A,5;
  D_2425 <= load B,5;
  D_2450 <= mul D_2443,D_2425;
  D_2451 <= add D_2449,D_2450;
  D_2452 <= mul D_2451,alpha;
  C <= store D_2452,4;
  D_2438 <= load A,3;
  D_2429 <= load B,6;
  D_2453 <= mul D_2438,D_2429;
  D_2440 <= load A,4;
  D_2431 <= load B,7;
  D_2454 <= mul D_2440,D_2431;
  D_2455 <= add D_2453,D_2454;
  D_2443 <= load A,5;
  D_2434 <= load B,8;
  D_2456 <= mul D_2443,D_2434;
  D_2457 <= add D_2455,D_2456;
  D_2458 <= mul D_2457,alpha;
  C <= store D_2458,5;
  D_2459 <= load A,6;
  D_2409 <= load B,0;
  D_2460 <= mul D_2459,D_2409;
  D_2461 <= load A,7;
  D_2412 <= load B,1;
  D_2462 <= mul D_2461,D_2412;
  D_2463 <= add D_2460,D_2462;
  D_2464 <= load A,8;
  D_2416 <= load B,2;
  D_2465 <= mul D_2464,D_2416;
  D_2466 <= add D_2463,D_2465;
  D_2467 <= mul D_2466,alpha;
  C <= store D_2467,6;
  D_2459 <= load A,6;
  D_2420 <= load B,3;
  D_2468 <= mul D_2459,D_2420;
  D_2461 <= load A,7;
  D_2422 <= load B,4;
  D_2469 <= mul D_2461,D_2422;
  D_2470 <= add D_2468,D_2469;
  D_2464 <= load A,8;
  D_2425 <= load B,5;
  D_2471 <= mul D_2464,D_2425;
  D_2472 <= add D_2470,D_2471;
  D_2473 <= mul D_2472,alpha;
  C <= store D_2473,7;
  D_2459 <= load A,6;
  D_2429 <= load B,6;
  D_2474 <= mul D_2459,D_2429;
  D_2461 <= load A,7;
  D_2431 <= load B,7;
  D_2475 <= mul D_2461,D_2431;
  D_2476 <= add D_2474,D_2475;
  D_2464 <= load A,8;
  D_2434 <= load B,8;
  D_2477 <= mul D_2464,D_2434;
  D_2478 <= add D_2476,D_2477;
  D_2479 <= mul D_2478,alpha;
  C <= store D_2479,8;
  L_EXIT <= jmpun;
L_EXIT:
  nop;
}
