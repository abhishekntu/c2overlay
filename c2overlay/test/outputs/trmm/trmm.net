.input N8
pinlist: N8

.input N9
pinlist: N9

.input N1
pinlist: N1

.input N2
pinlist: N2

.input N3
pinlist: N3

.input N4
pinlist: N4

.input N5
pinlist: N5

.input N6
pinlist: N6

.input N7
pinlist: N7

.input N12
pinlist: N12

.input N13
pinlist: N13

.input N10
pinlist: N10

.input N11
pinlist: N11

.input N16
pinlist: N16

.input N17
pinlist: N17

.input N14
pinlist: N14

.input N15
pinlist: N15

.input N18
pinlist: N18

.output out:N55
pinlist: N55

.output out:N58
pinlist: N58

.output out:N63
pinlist: N63

.output out:N62
pinlist: N62

.output out:N61
pinlist: N61

.output out:N60
pinlist: N60

.output out:N59
pinlist: N59

.output out:N56
pinlist: N56

.output out:N57
pinlist: N57

.clb N54_blk
pinlist: N12 N13 open open N54 open open open open 
subblock: N54_blk 0 1 open open 4 open open open open

.clb N49_blk
pinlist: N34 N14 N15 open N49 open open open open 
subblock: N49_blk 0 1 2 open 4 open open open open

.clb N48_blk
pinlist: N6 N9 open open N48 open open open open 
subblock: N48_blk 0 1 open open 4 open open open open

.clb N41_blk
pinlist: N22 open open open N58 open open open open 
subblock: N41_blk 0 open open open 4 open open open open

.clb N40_blk
pinlist: N2 N38 N10 open N40 open open open open 
subblock: N40_blk 0 1 2 open 4 open open open open

.clb N43_blk
pinlist: N1 N9 open open N43 open open open open 
subblock: N43_blk 0 1 open open 4 open open open open

.clb N42_blk
pinlist: N8 N9 N45 open N42 open open open open 
subblock: N42_blk 0 1 2 open 4 open open open open

.clb N45_blk
pinlist: N5 N16 open open N45 open open open open 
subblock: N45_blk 0 1 open open 4 open open open open

.clb N44_blk
pinlist: N6 N7 N46 open N44 open open open open 
subblock: N44_blk 0 1 2 open 4 open open open open

.clb N47_blk
pinlist: N23 open open open N57 open open open open 
subblock: N47_blk 0 open open open 4 open open open open

.clb N46_blk
pinlist: N3 N18 open open N46 open open open open 
subblock: N46_blk 0 1 open open 4 open open open open

.clb N29_blk
pinlist: N3 N13 open open N29 open open open open 
subblock: N29_blk 0 1 open open 4 open open open open

.clb N28_blk
pinlist: N1 N17 N54 open N28 open open open open 
subblock: N28_blk 0 1 2 open 4 open open open open

.clb N23_blk
pinlist: N24 N10 N15 open N23 open open open open 
subblock: N23_blk 0 1 2 open 4 open open open open

.clb N22_blk
pinlist: N2 N26 N4 open N22 open open open open 
subblock: N22_blk 0 1 2 open 4 open open open open

.clb N21_blk
pinlist: N20 open open open N61 open open open open 
subblock: N21_blk 0 open open open 4 open open open open

.clb N20_blk
pinlist: N25 N11 N14 open N20 open open open open 
subblock: N20_blk 0 1 2 open 4 open open open open

.clb N27_blk
pinlist: N32 N4 N11 open N27 open open open open 
subblock: N27_blk 0 1 2 open 4 open open open open

.clb N26_blk
pinlist: N5 N18 N48 open N26 open open open open 
subblock: N26_blk 0 1 2 open 4 open open open open

.clb N25_blk
pinlist: N1 N29 N7 open N25 open open open open 
subblock: N25_blk 0 1 2 open 4 open open open open

.clb N24_blk
pinlist: N31 N8 N17 open N24 open open open open 
subblock: N24_blk 0 1 2 open 4 open open open open

.clb N19_blk
pinlist: N27 open open open N62 open open open open 
subblock: N19_blk 0 open open open 4 open open open open

.clb N52_blk
pinlist: N51 open open open N63 open open open open 
subblock: N52_blk 0 open open open 4 open open open open

.clb N53_blk
pinlist: N36 open open open N55 open open open open 
subblock: N53_blk 0 open open open 4 open open open open

.clb N50_blk
pinlist: N12 N18 open open N50 open open open open 
subblock: N50_blk 0 1 open open 4 open open open open

.clb N51_blk
pinlist: N28 N10 N11 open N51 open open open open 
subblock: N51_blk 0 1 2 open 4 open open open open

.clb N38_blk
pinlist: N6 N17 N50 open N38 open open open open 
subblock: N38_blk 0 1 2 open 4 open open open open

.clb N39_blk
pinlist: N49 open open open N60 open open open open 
subblock: N39_blk 0 open open open 4 open open open open

.clb N30_blk
pinlist: N4 N42 N15 open N30 open open open open 
subblock: N30_blk 0 1 2 open 4 open open open open

.clb N31_blk
pinlist: N12 N16 open open N31 open open open open 
subblock: N31_blk 0 1 open open 4 open open open open

.clb N32_blk
pinlist: N5 N13 N43 open N32 open open open open 
subblock: N32_blk 0 1 2 open 4 open open open open

.clb N33_blk
pinlist: N30 open open open N59 open open open open 
subblock: N33_blk 0 open open open 4 open open open open

.clb N34_blk
pinlist: N3 N35 N16 open N34 open open open open 
subblock: N34_blk 0 1 2 open 4 open open open open

.clb N35_blk
pinlist: N7 N8 open open N35 open open open open 
subblock: N35_blk 0 1 open open 4 open open open open

.clb N36_blk
pinlist: N2 N14 N44 open N36 open open open open 
subblock: N36_blk 0 1 2 open 4 open open open open

.clb N37_blk
pinlist: N40 open open open N56 open open open open 
subblock: N37_blk 0 open open open 4 open open open open

