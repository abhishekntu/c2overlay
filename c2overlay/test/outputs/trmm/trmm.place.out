Netlist file: test.net   Architecture file: test.arch.xml
Array size: 7 x 7 logic blocks

#block name	x	y	subblk	block number
#----------	--	--	------	------------
N8		8	1	0	#0
N9		8	2	0	#1
N1		3	0	0	#2
N2		0	5	0	#3
N3		0	3	0	#4
N4		8	5	0	#5
N5		8	3	0	#6
N6		1	0	0	#7
N7		6	0	0	#8
N12		2	0	0	#9
N13		4	0	0	#10
N10		0	6	0	#11
N11		3	8	0	#12
N16		8	6	0	#13
N17		0	1	0	#14
N14		0	4	0	#15
N15		8	4	0	#16
N18		5	0	0	#17
out:N55		0	7	0	#18
out:N58		5	8	0	#19
out:N63		1	8	0	#20
out:N62		4	8	0	#21
out:N61		2	8	0	#22
out:N60		6	8	0	#23
out:N59		8	7	0	#24
out:N56		0	2	0	#25
out:N57		7	0	0	#26
N54_blk		4	1	0	#27
N49_blk		6	4	0	#28
N48_blk		5	2	0	#29
N41_blk		5	6	0	#30
N40_blk		1	3	0	#31
N43_blk		4	2	0	#32
N42_blk		7	2	0	#33
N45_blk		7	3	0	#34
N44_blk		2	2	0	#35
N47_blk		7	1	0	#36
N46_blk		2	3	0	#37
N29_blk		3	3	0	#38
N28_blk		3	1	0	#39
N23_blk		5	4	0	#40
N22_blk		5	5	0	#41
N21_blk		2	4	0	#42
N20_blk		3	4	0	#43
N27_blk		4	4	0	#44
N26_blk		5	3	0	#45
N25_blk		3	2	0	#46
N24_blk		5	1	0	#47
N19_blk		4	5	0	#48
N52_blk		2	5	0	#49
N53_blk		1	5	0	#50
N50_blk		2	1	0	#51
N51_blk		3	5	0	#52
N38_blk		1	1	0	#53
N39_blk		6	5	0	#54
N30_blk		7	4	0	#55
N31_blk		6	1	0	#56
N32_blk		4	3	0	#57
N33_blk		7	5	0	#58
N34_blk		6	3	0	#59
N35_blk		6	2	0	#60
N36_blk		1	4	0	#61
N37_blk		1	2	0	#62
